contracts/handlers/GenericHandler.sol

```diff
@@ -112,6 +115,14 @@ contract GenericHandler is IDepositHandler, ERC20Safe {
             )
         }
 
+        address contractAddress = _resourceIDToContractAddress[resourceID];
+        require(_contractWhitelist[contractAddress], "provided contractAddress is not whitelisted");
+
+        if (_contractAddressToDepositFunctionSignature[contractAddress] != bytes4(0)) {
```

Comment:
I think we should be using the signature from the mapping. I believe we can append it to metadata, and pass that to `call`. We want to make sure only the whitelisted function can be called, in the current state I think any method could be called